src/ConfigParser.cc

```diff
@@ -561,6 +562,20 @@ ConfigParser::rejectDuplicateDirective()
     throw TextException("duplicate configuration directive", Here());
 }
 
+SBuf
+ConfigParser::openDirective(const SBuf &line)
+{
+    Parser::Tokenizer tk(line);
+    SBuf directiveName; // TODO: Upgrade cfg_directive to SBuf and set it here.
+    static const auto spaceChars = CharacterSet("w_space", w_space);
+    static const auto directiveChars = spaceChars.complement("squid.conf directive name");
+    const auto found = tk.prefix(directiveName, directiveChars);
```

Comment:
This introduces a large number of future problems with UTF character encodings and invisible ASCII characters.

Please keep the parser restricted to just the set of characters we currently use in directive names. We can add other specials in future as-needed.
```suggestion
    static const auto directiveChars = CharacterSet("squid.conf directive name","_") + CharacterSet::DIGIT + CharacterSet::ALPHA;
    const auto found = tk.prefix(directiveName, directiveChars);
```

Reply:
> This introduces a large number of future problems with UTF character encodings and invisible ASCII characters.

This code does not _introduce_ those problems: "UTF character encodings and invisible ASCII characters" are handled essentially the same way by current master/v7 code that uses [strtok](https://en.cppreference.com/w/c/string/byte/strtok)() with the same w_space delimiters:

```C++
static int
parse_line(char *buff)
{
        char    *token;
        if ((token = strtok(buff, w_space)) == NULL) 
                return 1;       /* ignore empty lines */
        ConfigParser::SetCfgLine(strtok(nullptr, ""));
    if (!strcmp(token, "broken_vary_encoding"))
        ...
```

This PR _preserves_ that (arguably unwanted) behavior.

> Please keep the parser restricted to just the set of characters we currently use in directive names. We can add other specials in future as-needed.

s/keep/make/ -- the current official parser is not restricted that way.

Are you sure this change should be done in _this_ PR? I will do it if you insist, but it feels like it should be done in a followup PR that is focused on restricting input in these areas. I do support restricting input, but this PR is not about that.

Reply:
> Please keep the parser restricted to just the set of characters we currently use in directive names.

Also, the current directive name parser (in official code and in this PR) is already _effectively_ restricting directive names to "characters we currently use in directive names" -- the parser is written as a series of `if` statements matching recognized directives names and rejecting any other names (including any name with prohibited characters).

The only thing we can add here is a new explicit "[not only you are using an unrecognized directive name, but] you are using a directive name with an invalid character" error. Adding such an error requires more/different code than code suggested in this change request. Such an error will be useful in some rare cases, but adding such an error can clearly be viewed as being outside _this_ "SBuf upgrade" PR scope.